tools/hlsl_generator/gen.py

```diff
@@ -28,6 +28,25 @@
 namespace spirv
 {
 
+//! General Decls
+template<class T>
+NBL_CONSTEXPR_STATIC_INLINE bool is_pointer_v = is_spirv_type<T>::value;
+
+template<uint32_t StorageClass, typename T>
+struct pointer
+{
+   using type = vk::SpirvOpaqueType<spv::OpTypePointer, vk::Literal< vk::integral_constant<uint32_t, StorageClass> >, T>;
+};
+// partial spec for BDA
+template<typename T>
+struct pointer<spv::StorageClassPhysicalStorageBuffer, T>
+{
+   using type = vk::SpirvType<spv::OpTypePointer, sizeof(uint64_t), sizeof(uint64_t), vk::Literal<vk::integral_constant<uint32_t, spv::StorageClassPhysicalStorageBuffer> >, T>;
+};
+
+template<uint32_t StorageClass, typename T>
+using pointer_t = typename pointer<StorageClass, T>::type;
+
```

Comment:
only `is_` should be implemented via structs

Reply:
wdym? it's from the snippet you sent here https://github.com/Devsh-Graphics-Programming/SPIRV-Headers/pull/3#discussion_r1748922865

Reply:
sorry I'm dumb.

Still the `is_pointer_v` is wrong cause you're checking for being a spir-v type, not `OpTypePointer`